sdk/nodejs/service/index.ts

```diff
@@ -0,0 +1,19 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as runtime from "../runtime";
+
+export function getStack(name: string): Promise<any> {
```

Comment:
As usual - the fact that this returns a Promise is going to make it unfortunately painful to use in practice - you won't be able to do what I've done in the code examples above where you just dot into the result - you'll need `.then` to do the simplest things.

At the very least, that may argue for having a richer API that allows common things without `then` - like `pulumi.service.getStackOutput("lukehoban/app-production", "url")` whcih returns a Promise<string>.

I think we should write out some examples that really use this to make sure we like the developer experience.

Reply:
> As usual - the fact that this returns a Promise is going to make it unfortunately painful to use in practice - you won't be able to do what I've done in the code examples above where you just dot into the result - you'll need .then to do the simplest things.

That's right. And I agree that the experience isn't exactly what we want it to be, but at the very least it's consistent with our other invokes.

The only alternative of which I'm aware is to make this strictly an SDK-level construct. If we rely on a provider to implement this functionality, we cannot make this a synchronous call.

> I think we should write out some examples that really use this to make sure we like the developer experience.

Yep, that's a good idea. The primary challenge I see here is that the promise-based API makes it more difficult to make control-flow decisions based on another stack's outputs. Dataflow should be fine.

Reply:
> The only alternative of which I'm aware is to make this strictly an SDK-level construct. If we rely on a provider to implement this functionality, we cannot make this a synchronous call.

Out of curiosity - how were you imagining we could make this synchronous as a SDK-level construct?